scipy/sparse/linalg/_dsolve/_superluobject.c

```diff
@@ -105,10 +106,159 @@ static PyObject *SuperLU_solve(SuperLUObject * self, PyObject * args,
+        if (ord == -1 && PyErr_Occurred()) {
+            return NULL;  /* conversion error */
+        }
+        if (ord == 1) {
+            norm_c = '1';
+        } else {
+            PyErr_SetString(PyExc_ValueError, "ord must be 1 or np.inf");
+            return NULL;
+        }
+    } else if (PyFloat_Check(ord_obj)) {  /* check if it's a float (np.inf) */
+        double ord = PyFloat_AsDouble(ord_obj);
+        if (ord == -1.0 && PyErr_Occurred()) {
+            return NULL;  /* conversion error */
+        }
+        if (ord == NPY_INFINITY) {
+            norm_c = 'I';  /* infinity norm */
+        } else {
+            PyErr_SetString(PyExc_ValueError, "ord must be 1 or np.inf");
+            return NULL;
+        }
+    } else {
+        PyErr_SetString(PyExc_TypeError, "ord must be 1 or np.inf");
+        return NULL;
+    }
+
+    const bool return_float = (self->type == NPY_FLOAT || self->type == NPY_CFLOAT);
+
+    /* Check for empty matrix */
+    if (is_empty_matrix(&self->L) && is_empty_matrix(&self->U)) {
+        /* Empty matrix, return 0.0 */
+        if (return_float) {
+            float condf = 0.0f;
+            return PyArray_Scalar(&condf, PyArray_DescrFromType(NPY_FLOAT), NULL);
+        } else {
+            double condd = 0.0;
+            return PyArray_Scalar(&condd, PyArray_DescrFromType(NPY_DOUBLE), NULL);
+        }
+    }
+
+    jmpbuf_ptr = (volatile jmp_buf *)superlu_python_jmpbuf();
+    if (setjmp(*(jmp_buf*)jmpbuf_ptr)) {
+        goto fail;
+    }
+
+    /* Output from gscon */
+    volatile SuperLUStat_t stat = { 0 };
+    StatInit((SuperLUStat_t *)&stat);
+
+    jmpbuf_ptr = (volatile jmp_buf *)superlu_python_jmpbuf();
+
+    SLU_BEGIN_THREADS;
+
+    if (setjmp(*(jmp_buf*)jmpbuf_ptr)) {
+        SLU_END_THREADS;
+        goto fail;
+    }
+
+    /* Outputs from gscon */
+    volatile float rcondf;
+    volatile double rcondd;
+    volatile int info;
+
+    switch (self->type) {
+        case NPY_FLOAT:
+            sgscon(
+                (char *)&norm_c, &self->L, &self->U, self->anorm.f,
+                (float *)&rcondf, (SuperLUStat_t *)&stat, (int *)&info);
+            break;
+        case NPY_DOUBLE:
+            dgscon(
+                (char *)&norm_c, &self->L, &self->U, self->anorm.d,
+                (double *)&rcondd, (SuperLUStat_t *)&stat, (int *)&info);
+            break;
+        case NPY_CFLOAT:
+            cgscon(
+                (char *)&norm_c, &self->L, &self->U, self->anorm.f,
+                (float *)&rcondf, (SuperLUStat_t *)&stat, (int *)&info);
+            break;
+        case NPY_CDOUBLE:
+            zgscon(
+                (char *)&norm_c, &self->L, &self->U, self->anorm.d,
+                (double *)&rcondd, (SuperLUStat_t *)&stat, (int *)&info);
+            break;
+        default: 
+            PyErr_SetString(PyExc_ValueError, "unsupported data type");
+            return NULL;
+    }
+
+    SLU_END_THREADS;
+
+    if (info) {
```

Comment:
needs an explicit check for `info < 0` in case something goes weird and somehow info comes back positive overflow or whatever the case might be because ... C.